voxel51/platform/api.py

```diff
@@ -79,55 +79,58 @@ def close(self):
         if self.keep_alive:
             self._requests.close()
 
-    def get_job_data_urls(self, job_id):
+    def get_job_data_urls(self, task_config):
         '''Retrieves signed URLs to download job input data.
 
         Args:
-            job_id (str): the job ID
+            task_config (voxel51.platform.task.TaskConfig): the task config
 
         Returns:
             a dictionary mapping input names to RemotePathConfig objects
         '''
-        endpoint = self.base_url + "/jobs/" + job_id + "/url/data"
+        endpoint = self.base_url + "/jobs/" + task_config.job_id + "/url/data"
         res = self._requests.get(endpoint, headers=self._header)
-        _validate_response(res)
-        return {
-            k: voxu.RemotePathConfig(v)
-            for k, v in iteritems(_parse_json_response(res))
-        }
-
-    def get_job_status_url(self, job_id):
+        try:
+            _validate_response(res)
+            return {
+                k: voxu.RemotePathConfig(v)
+                for k, v in iteritems(_parse_json_response(res))
+            }
+        except:
```

Comment:
This should just catch `APIError`s, not everything

Reply:
These actually return HTTPErrors but, yes I agree this should be more specific

Reply:
_validate_response definitely returns an APIError: https://github.com/voxel51/platform-sdk/blob/df024d0fab67845477e121be53b3f46ad5dac09d/voxel51/platform/api.py#L245-L247
If you're seeing an HTTPError, [this](https://github.com/voxel51/platform-sdk/blob/df024d0fab67845477e121be53b3f46ad5dac09d/voxel51/platform/api.py#L235) is probably failing, which is an issue - do you have an example API response in this case?

Reply:
the response I was mocking up using the test server.. wonder if that caused this to occur..
```py
        if res.ok:
            raise ValueError("Response is not an error")
        try:
            message = _parse_json_response(res)["error"]["message"] <<- because this fails
        except ValueError:
            res.raise_for_status() <<<- this is where it was going, which creates an HHTPException based on the response code
        return cls(message, res.status_code)
```
I would change the server.js to:
```js
  function getSignedUrl(req, res) {
    const key = req.url.split('/').pop();
    recordEvent(SIGNED_URL_TO_EVENT[key], true);
    return {
      code: 500,
      //body: TASK[SIGNED_URL_TO_TASK_KEY[key] || key],
    };
  }
```
But this is likely not returning something with the fields "error" with the key "message"

... I see I should probably return an error like that too, and catch that APIError type as well

Reply:
Oh, yeah, I'd suggest testing against an API. The test server should match the API, though (and I thought it did, but commenting out the body like that probably wouldn't return the expected structure).
In any case, catching both HTTPError and APIError seems best.

Reply:
Yep, good point, tested with a well-formed APIError from a real API.  HTTPError will just be the fall back. Others should propagate.